mod/settings.php

```diff
@@ -381,24 +381,27 @@ function settings_post(App $a) {
 			$err = true;
         }
 
-        //  check if the old password was supplied correctly before
-        //  changing it to the new value
-        $r = q("SELECT `password` FROM `user`WHERE `uid` = %d LIMIT 1", intval(local_user()));
-        if( $oldpass != $r[0]['password'] ) {
-            notice( t('Wrong password.') . EOL);
-            $err = true;
-        }
+		//  check if the old password was supplied correctly before
+		//  changing it to the new value
+		$r = q("SELECT `password` FROM `user`WHERE `uid` = %d LIMIT 1", intval(local_user()));
+		if (!dbm::is_result($r)) {
+			killme();
+		} elseif ( $oldpass != $r[0]['password'] ) {
+			notice( t('Wrong password.') . EOL);
+			$err = true;
+		}
 
 		if(! $err) {
 			$password = hash('whirlpool',$newpass);
 			$r = q("UPDATE `user` SET `password` = '%s' WHERE `uid` = %d",
 				dbesc($password),
 				intval(local_user())
 			);
-			if($r)
+			if ($r) {
```

Comment:
The results of update queries should be checked via "dbm:is_result" as well, shouldn't it?